stablehlo/dialect/StablehloOps.cpp

```diff
@@ -635,6 +635,21 @@ mlir::Speculation::Speculatability DotGeneralOp::getSpeculatability() {
   return mlir::Speculation::Speculatable;
 }
 
+DotDimensionNumbersAttr getDefaultDotDimensionNumbers(mlir::Value lhs) {
+  return DotDimensionNumbersAttr::get(
+      lhs.getContext(),
+      /*lhsBatchingDimensions=*/{},
+      /*rhsBatchingDimensions=*/{},
+      /*lhsContractingDimensions=*/
+      {cast<ShapedType>(lhs.getType()).getRank() - 1},
+      /*rhsContractingDimensions=*/{0});
+}
+
+bool DotGeneralOp::isSimpleDot() {
```

Comment:
Attrs are stored in a context for the context's lifetime, can we implement this method without the additional DotDimAttr overhead?

If you see benefits to implementing this way then we can keep, the overhead is just one of these will live in the context forever, since there's good caching.

Reply:
My main concern was avoiding duplication. If we do all the checks explicitly, we are duplicating the knowledge of what it means to be a "simple dot" across two methods. Since the default attr is mostly empty, the memory overhead should not be too big? I think the overhead should be acceptable, but thanks for pointing this out!